File: algorithms/ar-isrt/python3/insertionSort.py
def insertionSort(array):
    n=len(array)
    i=0
    j=0
    for i in range(1,n):
        temp=array[i]
        print(temp)
        j=i
        while(j>0 and temp<array[j-1]):
            array[j]=array[j-1];
            j-=1
        array[j]=temp
    return array

a=[5,2,4,3,0,6,7,8,11]
print(insertionSort(a))

                   

File: algorithms/ar-isrt/csharp/InsertionSort.cs
// C# program to implement Insertion Sort 
using System; 

public static class InsertionSort { 

	// Function to sort array using insertion sort 
	static void sort(int[] arr) 
	{ 
		int n = arr.Length; 
		for (int i = 1; i < n; ++i) 
		{ 
			int key = arr[i]; 
			int j = i - 1; 


			while (j >= 0 && arr[j] > key) 
			{ 
				arr[j + 1] = arr[j]; 
				j = j - 1; 
			} 
			arr[j + 1] = key; 
		} 
	} 

	// Function to print array of size n 
	static void printArray(int[] arr) 
	{ 
		int n = arr.Length; 
		for (int i = 0; i < n; ++i) 
		{
			//print(arr[i]) 
		}

	} 

	// Driver Code to test above function
	public static void Main() 
	{ 
		int[] arr = { 12, 11, 13, 5, 6 , 1, 2, 5, 4, 7, 8, 6, 8}; 
		sort(arr); 
		printArray(arr); 
	} 
} 

